liota/dccs/iotcc.py

```diff
@@ -398,28 +401,37 @@ def set_properties(self, reg_entity_obj, properties):
             # get dev_type, and prop_dict if possible
             with self.file_ops_lock:
                 self._store_reg_entity_attributes("Devices", entity, reg_entity_obj.reg_entity_id,
-                                                 entity.entity_type, properties)
+                                                  entity.entity_type, properties)
 
     def publish_unit(self, reg_entity_obj, metric_name, unit):
         """
          Publish SI units as properties for Metrics but RegisteredMetric object are simply returned
          so currently units are set to the parent RegisteredEntity(Device or EdgeSystem)
+         Either units are set with prefix as properties or only unit gets set as property if prefix doesn't exist.
         :param reg_entity_obj: RegisteredEntity Object
         :param metric_name: Metric Name
         :param unit: SI Unit
         :return:
         """
         str_prefix, str_unit_name = parse_unit(unit)
-        if not isinstance(str_prefix, basestring):
-            str_prefix = ""
-        if not isinstance(str_unit_name, basestring):
-            str_unit_name = ""
-        properties_added = {
-            metric_name + "_unit": str_unit_name,
-            metric_name + "_prefix": str_prefix
-        }
-        self.set_properties(reg_entity_obj, properties_added)
-        log.info("Published metric unit with prefix to IoTCC")
+        if not isinstance(str_prefix, basestring) and isinstance(str_unit_name, basestring):
+            properties_unit_list = {
+                metric_name + "_unit": str_unit_name
+            }
+            log.debug("Publishing unit {0} for metric {1} to IoTCC".format(str_unit_name, metric_name))
```

Comment:
It will also be good to get the entity name (and id?) in the log, as different entities can have the same metric name

Reply:
Entity_name gets printed in the set_properties call which is called from "publish_unit" method. Hence, I don't think it will be required to be added here.